lib/settings.py

```diff
@@ -212,47 +190,50 @@ def prompt(question, choices):
+    with open("{}/lib/data_files/wordlist_links".format(os.getcwd())) as wordlist_links:
+        if multi == 1:
+            b64link = random.choice(wordlist_links.readlines())
+            filename = "Download-" + random_salt_generator(use_string=True)[0]
+            LOGGER.info("Beginning download..")
+            with open("{}.txt".format(filename), "a+") as wordlist:
+                response = requests.get(base64.b64decode(b64link.strip()), stream=True)
+                total_length = response.headers.get('content-length')
                 if verbose:
-                    LOGGER.debug("Starting download at {}..".format(start))
-                downloaded = 0
-                total_length = int(total_length)
-                for data in response.iter_content(chunk_size=1024):
-                    downloaded += len(data)
-                    wordlist.write(data)
-                    done = int(50 * downloaded / total_length)
-                    sys.stdout.write("\r[\033[93m{}\033[0m{}]".format("#" * done, " " * (50-done)))
-                    sys.stdout.flush()
-        print("")
-        LOGGER.info("Download complete, saved under: {}.txt. Time elapsed: {}s".format(filename, time.time() - start))
-    else:
-        if multi <= len(WORDLIST_LINKS):
-            for _ in range(int(multi)):
-                LOGGER.info("Downloading wordlist #{}..".format(_ + 1))
-                download_rand_wordlist(verbose=verbose)
+                    LOGGER.debug("Content length to be downloaded: {}..".format(convert_file_size(int(total_length))))
+                    LOGGER.debug("Wordlist link downloading from: '{}'..".format(b64link.strip()))
+                if total_length is None:
+                    wordlist.write(response.content)
+                else:
+                    start = time.time()
+                    if verbose:
+                        LOGGER.debug("Starting download at {}..".format(start))
+                    downloaded = 0
+                    total_length = int(total_length)
+                    for data in response.iter_content(chunk_size=1024):
+                        downloaded += len(data)
+                        wordlist.write(data)
+                        done = int(50 * downloaded / total_length)
+                        sys.stdout.write("\r[\033[93m{}\033[0m{}]".format("#" * done, " " * (50-done)))
+                        sys.stdout.flush()
+            print("")
+            LOGGER.info("Download complete, saved under: {}.txt. Time elapsed: {}s".format(filename, time.time() - start))
         else:
-            LOGGER.fatal(
-                "There are currently {} wordlists available for download, and you "
-                "have chose to download {}. Seeing as there aren't that many right now "
-                "this download will fail, try again with a smaller number..".format(len(WORDLIST_LINKS), multi)
-            )
+            if multi <= len(wordlist_links.readlines()):
```

Comment:
This part here is making me think of how to account for that using the `click` cli library. I read this section the other day and it seems like it might fit perfectly for this situation.

http://click.pocoo.org/5/options/#range-options

> an optional clamping mode where a value that falls outside of the range will be clamped. This means that a range of 0-5 would return 5 for the value 10 or 0 for the value -1 (for example).

`@click.option('--count', type=click.IntRange(0, 20, clamp=True))`

The `IntRange` could be set at however many wordlists there are and then if a given number is too large it would "clamp" it to the maximum and if the given number is too small it would "clamp" it to the minimum.

I actually rewrote most of cli interface with `click` the other day. It's more of a prototype right now though, many updates needed.